plugins/dell-dock/fu-dell-dock-i2c-ec.c

```diff
@@ -968,16 +954,13 @@ fu_dell_dock_ec_open (FuDevice *device, GError **error)
 static gboolean
 fu_dell_dock_ec_close (FuDevice *device, GError **error)
 {
-	FuDellDockEc *self = FU_DELL_DOCK_EC (device);
-
-	return fu_device_close (self->symbiote, error);
+	return fu_device_close (fu_device_get_proxy (device), error);
 }
 
 static void
 fu_dell_dock_ec_finalize (GObject *object)
 {
 	FuDellDockEc *self = FU_DELL_DOCK_EC (object);
-	g_object_unref (self->symbiote);
```

Comment:
I guess this is actually a bug from previously, I didn't see it actually refcounted previously.

Reply:
I do wonder if it's safer to actually ref it on open though, no?

Reply:
I don't think we can ref it, else we get a lifecycle dep loop and the device never finalizes.

Reply:
Then maybe we shouldn't be doing that in the i2c TBT code too.

Reply:
Ohh, i missed that one. Will fix.

Reply:
Fixed, and added the GUID thing we needed to avoid the MST hack. The GUID engine matcher thing also allows me to fix the VLI PD update thing I think.